### PR TITLE
Add persist client generation support for Redis DB

### PR DESCRIPTION
**Pull request by dinukaamarasinghe817**
## Purpose
To generate the persist client code automatically for Redis DB

Fixes: https://github.com/ballerina-platform/persist-tools/issues/328

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_100_redis/Config.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[persist_generate_100.entities.redis]
+connection = "redis://localhost:6379"
+
+
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_100_redis/Config.toml:4]:
```suggestion
```

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_96_redis/persist/model.bal [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 202e WSO2 LLC. (http://www.wso2.org).
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_96_redis/persist/model.bal:1]:
```suggestion
// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.org).
```

**Comment by dinukaamarasinghe817** [comment]
> @dinukaamarasinghe817 Shall we add an example also Redis DB?

I have added an example scenario for Redis Db in here https://github.com/ballerina-platform/persist-tools/pull/333/commits/a47e9c401642d1ee32077e4b83f1420855854ec4

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_96_redis/persist/model.bal (hunk 2 above)
Comment (dinukaamarasinghe817) [reply persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_96_redis/persist/model.bal]: This has been fixed in https://github.com/ballerina-platform/persist-tools/pull/333/commits/a47e9c401642d1ee32077e4b83f1420855854ec4

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_100_redis/Config.toml (hunk 1 above)
Comment (dinukaamarasinghe817) [reply persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_100_redis/Config.toml]: This has been fixed in https://github.com/ballerina-platform/persist-tools/pull/333/commits/a47e9c401642d1ee32077e4b83f1420855854ec4

File: examples/social_media/persist/model.bal [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+public type User record{|
+    readonly string firstName;
+    readonly string lastName;
+    string email;
+    int age;
+    time:Date dateOfBirth;
+    Gender gender;
+    boolean isMarried;
+    string spouseName?;
+	Post[] post;
```
Comment (daneshk) [inline examples/social_media/persist/model.bal:34]:
```suggestion
    string spouseName?;
    Post[] post;
```

File: examples/social_media/persist/model.bal [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+|};
+
+public type Post record{|
+    readonly int id;
+    string title;
+    string content;
+    User user;
+	Comment[] comment;
```
Comment (daneshk) [inline examples/social_media/persist/model.bal:42]:
```suggestion
    User user;
    Comment[] comment;
```

Comment (dinukaamarasinghe817) [reply examples/social_media/persist/model.bal]: Fixed in https://github.com/ballerina-platform/persist-tools/pull/333/commits/32874f0238f447941f8a9e3ebfae4872414ca2bb

File: examples/social_media/persist/model.bal (hunk 3 above)
Comment (dinukaamarasinghe817) [reply examples/social_media/persist/model.bal]: Fixed in https://github.com/ballerina-platform/persist-tools/pull/333/commits/32874f0238f447941f8a9e3ebfae4872414ca2bb

**Review by daneshk — APPROVED** [review]

**Pull request merged**